### PR TITLE
Link batch results on /tide to deck page.

### LINKED ISSUE #7458: Link batch in tide frontend  back to the batch in the main deck page (Prow Status)
Something convenient that the SQ had and is missing from tide.

/area prow

@BenTheElder @stevekuznetsov @cjwagner

### PR DESCRIPTION
**Pull request by rmmh**
Fixes #7458.

image
image

### CODE REVIEW

**Comment by rmmh** [comment]
ping

File: prow/cmd/deck/static/tidescript.js [hunk 1]
```diff
@@ -227,6 +227,21 @@ function createPRCell(pool, prs) {
...
+    var td = document.createElement('td');
+    if (pool.BatchPending) {
+        var numbers = pool.BatchPending.map(p => String(p.Number));
+        var batchRef = pool.Branch + ',' + numbers.join(',');
+        var href = '/?repo=' + encodeURIComponent(pool.Org + '/' + pool.Repo) +
+            '&type=batch&pull=' + encodeURIComponent(batchRef);
+        var link = document.createElement('a');
+        link.href = href;
+        link.appendChild(document.createTextNode(numbers.join(' ')));
+        td.appendChild(link, td.firstChild);
```
Comment (cjwagner) [inline prow/cmd/deck/static/tidescript.js]: What does the second argument do? I can't find a reference for this version of `appendChild`.

File: prow/cmd/deck/static/script.js [hunk 2]
```diff
@@ -16,13 +16,18 @@ function updateQueryStringParameter(uri, key, value) {
     }
 }
 
+function shortenBatchRef(build) {
+    return build.refs.replace(/:[0-9a-f]*/g, '');
```
Comment (cjwagner) [inline prow/cmd/deck/static/script.js]: Is the first argument a valid string?

File: prow/cmd/deck/static/tidescript.js [hunk 3]
```diff
@@ -227,6 +227,21 @@ function createPRCell(pool, prs) {
     return c;
 }
 
+function createBatchCell(pool) {
+    var td = document.createElement('td');
+    if (pool.BatchPending) {
```
Comment (cjwagner) [inline prow/cmd/deck/static/tidescript.js]: `[]` is truthy so this should probably be `if (pool.BatchPending && pool.BatchPending != [])`.

**Pull request merged**